frontend/lib/src/components/shared/FullScreenWrapper/FullScreenWrapper.tsx

```diff
@@ -46,131 +43,66 @@ export interface FullScreenWrapperProps {
-    const paddingTop = convertScssRemValueToPixels(
-      this.props.theme.sizes.fullScreenHeaderHeight
-    )
-
-    return {
-      fullWidth: window.innerWidth - padding * 2, // Left and right
-      fullHeight: window.innerHeight - (padding + paddingTop), // Bottom and Top
-    }
-  }
-
-  private updateWindowDimensions = (): void => {
-    this.setState(this.getWindowDimensions())
+const FullScreenWrapper: FC<FullScreenWrapperProps> = ({
+  children,
+  width,
+  height,
+  disableFullscreenMode,
+}) => {
+  const { expanded, fullHeight, fullWidth, zoomIn, zoomOut } = useFullscreen()
+
+  let buttonImage = FullscreenEnter
+  let buttonOnClick = zoomIn
+  let buttonTitle = "View fullscreen"
+
+  if (expanded) {
+    buttonImage = FullscreenExit
+    buttonOnClick = zoomOut
+    buttonTitle = "Exit fullscreen"
   }
 
-  public render(): JSX.Element {
-    const { expanded, fullWidth, fullHeight } = this.state
-    const { children, width, height, disableFullscreenMode } = this.props
-
-    let buttonImage = FullscreenEnter
-    let buttonOnClick = this.zoomIn
-    let buttonTitle = "View fullscreen"
-
-    if (expanded) {
-      buttonImage = FullscreenExit
-      buttonOnClick = this.zoomOut
-      buttonTitle = "Exit fullscreen"
-    }
-
-    return (
-      <StyledFullScreenFrame
-        isExpanded={expanded}
-        data-testid="stFullScreenFrame"
-      >
-        {!disableFullscreenMode && (
-          <StyledFullScreenButton
-            data-testid="StyledFullScreenButton"
-            onClick={buttonOnClick}
-            title={buttonTitle}
-            isExpanded={expanded}
-          >
-            <Icon content={buttonImage} />
-          </StyledFullScreenButton>
-        )}
-        {expanded
-          ? children({
-              width: fullWidth,
-              height: fullHeight,
-              expanded,
-              expand: this.zoomIn,
-              collapse: this.zoomOut,
-            })
-          : children({
-              width,
-              height,
-              expanded,
-              expand: this.zoomIn,
-              collapse: this.zoomOut,
-            })}
-      </StyledFullScreenFrame>
-    )
-  }
+  return (
+    <StyledFullScreenFrame
+      isExpanded={expanded}
+      data-testid="stFullScreenFrame"
+    >
+      {!disableFullscreenMode && (
+        <StyledFullScreenButton
+          data-testid="StyledFullScreenButton"
+          onClick={buttonOnClick}
+          title={buttonTitle}
+          isExpanded={expanded}
+        >
+          <Icon content={buttonImage} />
+        </StyledFullScreenButton>
+      )}
+      {expanded
+        ? children({
+            width: fullWidth,
+            height: fullHeight,
+            expanded,
+            expand: zoomIn,
+            collapse: zoomOut,
+          })
+        : children({
+            width,
+            height,
+            expanded,
+            expand: zoomIn,
+            collapse: zoomOut,
+          })}
```

Comment:
nit: i know this was just copied, but wdyt about this simplification to clearly show only the width and height are changing?


```suggestion
      {children({
        width: expanded ? fullWidth : width,
        height: expanded ? fullHeight : height,
        expanded,
        expand: zoomIn,
        collapse: zoomOut,
      })}
```